stackit/internal/services/iaas/networkarea/resource.go

```diff
@@ -0,0 +1,678 @@
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"default_prefix_length": schema.Int64Attribute{
+				Description: "The default prefix length for networks in the network area.",
+				Optional:    true,
+				Computed:    true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(24),
+					int64validator.AtMost(29),
+				},
+				Default: int64default.StaticInt64(25),
+			},
+			"max_prefix_length": schema.Int64Attribute{
+				Description: "The maximal prefix length for networks in the network area.",
+				Optional:    true,
+				Computed:    true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(24),
+					int64validator.AtMost(29),
+				},
+				Default: int64default.StaticInt64(29),
+			},
+			"min_prefix_length": schema.Int64Attribute{
+				Description: "The minimal prefix length for networks in the network area.",
+				Optional:    true,
+				Computed:    true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(22),
+					int64validator.AtMost(29),
+				},
+				Default: int64default.StaticInt64(24),
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *networkAreaResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) { // nolint:gocritic // function signature required by Terraform
+	// Retrieve values from plan
+	var model Model
+	diags := req.Plan.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	organizationId := model.OrganizationId.ValueString()
+	ctx = tflog.SetField(ctx, "organization_id", organizationId)
+
+	// Generate API request body from model
+	payload, err := toCreatePayload(ctx, &model)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating network area", fmt.Sprintf("Creating API payload: %v", err))
+		return
+	}
+
+	// Create new network area
+	var httpResp *http.Response
+	ctxWithHTTPResp := runtime.WithCaptureHTTPResponse(ctx, &httpResp)
+	area, err := r.client.CreateNetworkArea(ctxWithHTTPResp, organizationId).CreateNetworkAreaPayload(*payload).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating network area", fmt.Sprintf("Calling API: %v", err))
+		return
+	}
+
+	networkArea, err := wait.CreateNetworkAreaWaitHandler(ctx, r.client, organizationId, *area.AreaId).WaitWithContext(context.Background())
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating network area", fmt.Sprintf("Network area creation waiting: %v", err))
+		return
+	}
+	networkAreaId := *networkArea.AreaId
+	ctx = tflog.SetField(ctx, "network_area_id", networkAreaId)
+
+	networkAreaRangesResp, err := r.client.ListNetworkAreaRanges(ctx, organizationId, networkAreaId).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error reading network area ranges", fmt.Sprintf("Calling API for network area  data: %v", err))
+		return
+	}
```

Comment:
Is this extra call needed? I think the network area ranges already come in the response of `CreateNetworkArea`, in this case they will be under `networkArea`